### PR TITLE
[webpack-node-external] Indicate the types are compatible with v3

### PR DESCRIPTION
**Pull request by mxdvl**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/liady/webpack-node-externals/compare/v2.5.2..v3.0.0
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/webpack-node-externals/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for webpack-node-externals 2.5
+// Type definitions for webpack-node-externals 3.0
```
Comment (peterblazejewicz) [inline types/webpack-node-externals/index.d.ts:1]: please bump minimum TS version to 4.4 (L.7):
```ts
// TypeScript Version: 4.4
```
this should fix the CI errors

Comment (mxdvl) [reply types/webpack-node-externals/index.d.ts:1]: Thanks for the tip, I tried a few versions locally to no avail. TS4.3 & up produce errors about `buffer` types being required but not part of the dependencies. Should `@types/node` be added as a peer?

Comment (mxdvl) [reply types/webpack-node-externals/index.d.ts:1]: I’ve gone in and implemented the same “fix” as https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64081#pullrequestreview-1269496629

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@mxdvl thanks for sticking with topic!

**Comment by mxdvl** [comment]
Ready to merge

**Pull request merged**